features/0420-rich-schemas-common/README.md

```diff
@@ -0,0 +1,251 @@
+```
+Writes a Rich Schema object to the ledger.
+
+#Params
+submitter: information about submitter
+data: {
+    id: Rich Schema object's ID (as a DID for example),
+    content: Rich Schema object as JSON-LD string,
+    rs_name: Rich Schema object name
+    rs_version: Rich Schema object version
+    rs_type: Rich schema object type
+    ver: the version of the generic object template
+},
+registry: identifier for the registry
+
+#Returns
+registry_response: result as json,
+error: {
+    code: aries common error code,
+    description:  aries common error description
+}
+```
+
+##### read_rich_schema_object
+```
+Reads a Rich Schema object from the ledger.
+
+#Params
+submitter (optional): information about submitter
+data: {
+    id: Rich Schema object's ID (as a DID for example),
+    ver: the version of the generic object template
+},
+registry: identifier for the registry
+
+#Returns
+registry_response: result as json,
+error: {
+    code: aries common error code,
+    description:  aries common error description
+}
+```
+
+## Reference
+- [0250: Rich Schema Objects](https://github.com/hyperledger/aries-rfcs/tree/master/concepts/0250-rich-schemas) 
+
+## Drawbacks
+
+Rich schema objects introduce more complexity.
+
+## Rationale and alternatives
+
+#### Rich Schema object ID
+The following options on how a Rich Schema object can be identified exist:
+- DID unique for each Rich Schema
+- DID URL with the origin (issuer's) DID as a base
+- DID URL with a unique (not issuer-related) DID as a base
+- UUID or other unique ID
+
+UUID doesn't provide global resolvability. We can not say what ledger the Rich Schema object belongs to 
+by looking at the UUID.
+
+DID and DID URL give persistence, global resolvability and decentralization.
+We can resolve the DID and understand what ledger the Rich Schema object belongs to.
+Also we can see that the object with the same id-string on different ledger is the same object (if id-string is calculated
+against a canonicalized hash of the content).  
+
+However, Rich Schema's DIDs don't have cryptographic verifiability property of common DIDs,
+so this is a DID not associated with keys in general case.
+This DID belongs neither to person nor organization nor thing.
+
+Using Issuer's DID (origin DID) as a base for DID URL may be too Indy-specific as other ledgers may not have
+an Issuer DID. Also it links a Rich Schema object to the Issuer belonging to a particular ledger. 
+
+So, we are proposing to use a unique DID for each Rich Schema object as it gives more natural way to identify 
+an entity in the distributed ledger world.  
+
+
+#### Rich Schema object as DID DOC
+If Rich Schema objects are identified by a unique DID, then a natural question is whether each Rich Schema object 
+needs to be presented as a DID DOC and resolved by a DID in a generic way.
+
+We are not requiring to define Rich Schema objects as DID DOCs for now. We may re-consider this in future once DID DOC format
```

Comment:
It would be good to better understand the linking between DIDs and the Rich Schema objects.

Reply:
Any Rich Schema object needs to have a unique ID.
We are proposing to use a DID as this unique ID. It may sounds a bit unnatural to do this as DIDs are usually associated with people, organizations or things possessing some keys and endpoints, but using DIDs for Rich Schemas looks like a convenient way to identify them on different blockchains and ledgers.
In general, whether to use DIDs for Rich Schema objects identification is an open question, and we would like to hear more cons and pros.